src/backend/RecentImagesPathStore.cpp

```diff
@@ -20,8 +20,15 @@
 #include "RecentImagesPathStore.h"
 
 
-RecentImagesPathStore::RecentImagesPathStore()
-	: mSettingsGroupPrefix("recentImagesPath"),
+RecentImagesPathStore &RecentImagesPathStore::instance()
+{
+	static RecentImagesPathStore _instance;
```

Comment:
Maybe we should move away from singleton, I've been thinking about it for the config too. Singeltons make testing harder. I would've gone probably further here and wrapped the QSetting in a wrapper class that has an interface and passed it in this class via constructor. Then you could in the tests just create a mock that implements that interface and check what was called, without creating real settings. Also it would make changing the underlying save method (how and where do we store the paths) easier to replace. I know it's a lot of additionally work but I think it will pay off later.

Reply:
Working on it.

Reply:
Tried making this change but it would require so many other changes; mainly because of the KsnipConfigProvider::instance() call. I would skip it TBH.